Added Calendar using react-big-calendar

Palaksharma23 opened the pull request:
Fixes #163

## Description
I have added a Calendar using react-big-calendar

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).

![image](https://user-images.githubusercontent.com/87171452/145710335-222e1593-59f1-4755-baba-eb5745a0817e.png)
![image](https://user-images.githubusercontent.com/87171452/145710337-02a70e1b-1b51-48a1-b311-3d371749b86b.png)

Palaksharma23 commented:
@sohamsshah can you please once review my pull request

HarshCasper reviewed (changes requested):
Hi @Palaksharma23 👋 

Exceptional work and effort! 

A few initial comments from my side: 

- Lint the codebase to mitigate linting issues and errors.
- Use a JSON file to have all the events (including past and upcoming events)
- Use something like [Luxon](https://moment.github.io/luxon/#/) to allow folks across different time zones to understand the meeting times with respect to their clock.
- Use UTC timezones throughout for uniformity. 
- Test for responsiveness. 

Tagging, @sohamsshah, our UI guru for a more detailed feedback. Great work 👏

Palaksharma23 commented:
Thank you @HarshCasper sir 😀
I will make commits for json file event management , UTC timezone and linting issues. 
But i will have to read about Luxon :)
I will do it asap

sohamsshah commented on website/src/pages/initiatives.js:
```diff
@@ -1,10 +1,45 @@
+  locales,
+});
+// format for adding a meeting is new Date(year , month-1 , date)
+const events = [
```
The events array must have following fields:
```
{
    title: 'Documentation Meeting',
    start: new Date(2021, 11, 2, 16, 30, 0), // should be having hour and minutes along with the time
    end: new Date(2021, 11, 2, 17, 0, 0),
    googleCalendarLink: 'link-to-google-calnedar', // onClick: this should take the user to the Google Calendar of the Moja Global Meetings
  },
  ```
  The events array should be placed `websiteData/meetings.js` and the data can be safely replaced with new ones. The array can be imported in the `initiatives.js` component.

sohamsshah commented on website/src/pages/initiatives.js:
```diff
@@ -71,37 +106,34 @@ const Community = () => {
+        <h3 className={styles.communitySubtitle}>Our Meetings</h3>
+           <div>
+            <div> 
               <h5 className={styles.meetingTitle}>Past Meetings</h5>
```
We no longer need the `Past Meetings` section. Only one calendar for the Meetings should be enough to track them.

sohamsshah commented on website/src/pages/initiatives.js:
```diff
@@ -71,37 +106,34 @@ const Community = () => {
-                {MeetingCard}
-              </List>
+                
+                  <Calendar className={styles.Calendar}
```
The Styles currently fail in `Dark Mode` toggle and the I feel that color schemes and contrast can be improved. Small UI changes can be implemented to make it look even better.

sohamsshah commented on website/src/pages/initiatives.js:
```diff
@@ -71,37 +106,34 @@ const Community = () => {
-              </List>
+                
+                  <Calendar className={styles.Calendar}
+                    localizer={localizer}
```
The Props which `<Calendar />` component shall accept:
```
<Calendar 
  className={styles.Calendar}
  localizer={localizer}
  events={events}
  selectable={true}
  views={['month', 'week', 'agenda']} // max number of meetings per day is 1, so it doesn't make sense to have day view
  onSelectEvent={(event) => redirect(event.googleCalendarLink)} // onSelectEvent performs an action when clicked on the event in the calendar
  style={{ height: 400, width: '100%' }} // can be altered based on the implemented style
/>
  ```

sohamsshah reviewed (changes requested):
Hey @Palaksharma23 , this is really appreciable work! 👏
I would suggest you to take time, understand and brain storm upon the review comments and focus on creating/solving with cleaner UI similar to [layer5 calendar](https://layer5.io/community/calendar). We don't want that fancy, but it is better to take time and create better UI so it requires less iterations for the implemented feature.

Once the changes are successfully added, we can safely delete the `PastMeetingCard` component and all related imports in the codebase.

As of now, the meetings section (past and upcoming) consisting of tables is now replaced with `React Big Calendar` with events listed in calendar form, which is definitely a good intuitive UI practice.

Thanks for working on it. Please let me know if you need any help in implementation or doubts for this.

Palaksharma23 commented:
thank you sir , i will do that @sohamsshah

Palaksharma23 commented:
@sohamsshah  sir can you please once review my last commit. 
i have added an add to calendar button which will link to google calendar of moja global community website , i have now provided a dummy link because i didn't have google calendar link and also i have added a double click functionality to the events of calendar which will also lead to google calendar of moja global community website

Palaksharma23 commented:
https://user-images.githubusercontent.com/87171452/146358906-a91bf991-6bc4-4ee3-8ad0-0b5e5a257b9e.mp4

Palaksharma23 commented:
https://user-images.githubusercontent.com/87171452/146359084-7a4b649e-b004-4794-a8a8-b8b1aac5917f.mp4

sohamsshah reviewed (approved):
LGTM @Palaksharma23 ! Thanks for working on this! 🎉

Palaksharma23 commented:
@sohamsshah  thank you sir , sure i will remove the merge conflicts

Palaksharma23 commented:
Done sir @ sohamshah :) 
Thanku so much for helping me with css ☺

sohamsshah commented:
> Done sir @sohamsshah :) Thanku so much for helping me with css ☺

Thanks for working on this. Appreciable work! 🔥
